### ISSUE TITLE
using the sessions store name as application is raising collision

### ISSUE THREAD

**Issue by TheLooseCannon** [simplabs/ember-simple-auth#2142]
i have a sessions store declared which is named application.ts and typescript is throwing a warning

```
WARNING: Detected collisions between .js and .ts files of the same name. This can result in nondeterministic build output; see https://git.io/JvIwo for more information.
  - hexnode-analytics/routes/application.{js,ts}
  - hexnode-analytics/session-stores/application.{js,ts}
```

and this at times spoils my build. can any name be used for the session store ? what all can be a possible alternative to this

**Comment by marcoow** [comment]
This is a problem with the combination of Ember CLI's tree-merging and using TypeScript. Ember CLI merges all of an application's addons' trees into the main app tree so that files in the app tree override files from the addon trees. The problem here is that ESA comes with a [standard session store](https://github.com/simplabs/ember-simple-auth/blob/master/app/session-stores/application.js) as well as a placeholder [`application` route](https://github.com/simplabs/ember-simple-auth/blob/master/app/routes/application.js) (which is needed in case you do not define the application route inside of your app so that ESA can [hook up its session restoration](https://github.com/simplabs/ember-simple-auth/blob/master/addon/initializers/setup-session-restoration.js)). These files are called `session-stores/application.js` and `routes/application.js` respectively. Usually, you would override those files by simply creating them inside of your app so they would replace ESA's files when the trees are merged. In your case, as you're using TypeScript though, your `.ts` will not actually replace ESA's files simply because they have different names and you end up with 2 files instead.

So long story short, you cannot use TypeScript for any files that you want to override files that are merged into your app tree from any addon trees. I don't actually think this is something that should be fixed in ESA at all but it's likely something that Ember CLI or ember-cli-typescript would have to address as it's more a general problem.

**Comment by marcoow** [comment]
closing since there's nothing we can do about this
